Code/Source/Camera/CameraSensor.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include "CameraSensor.h"
+
+#include <AzCore/Math/MatrixUtils.h>
+
+#include <Atom/RPI.Public/RPISystemInterface.h>
+#include <Atom/RPI.Public/Scene.h>
+#include <Atom/RPI.Public/RenderPipeline.h>
+#include <Atom/RPI.Public/Base.h>
+
+#include <AzFramework/Components/TransformComponent.h>
+#include <AzFramework/Scene/SceneSystemInterface.h>
+
+#include <Atom/RPI.Public/Pass/Specific/RenderToTexturePass.h>
+
+#include <PostProcess/PostProcessFeatureProcessor.h>
+
+#include <Camera/CameraCaptureScheduler.h>
+
+namespace ROS2 {
+    CameraSensorDescription::CameraSensorDescription(const AZStd::string& cameraName, float verticalFov, int width, int height)
+    : verticalFieldOfViewDeg(verticalFov)
+    , width(width)
+    , height(height)
+    , cameraName(cameraName)
+    , aspectRatio(static_cast<float>(width)/static_cast<float>(height))
+    , viewToClipMatrix(MakeViewToClipMatrix()) {
+        validateParameters();
+    }
+
+    AZ::Matrix4x4 CameraSensorDescription::MakeViewToClipMatrix() const {
+        const float nearDist = 0.1f, farDist = 100.0f;
```

Comment:
In the future, this might be an (advanced) configurable parameter of the component, I'd leave a TODO here.

Reply:
I treated it as an internal rendering parameter and not for the user to decide. As an advanced property maybe, yes but I do not see the point of marking it with TODO. If such a necessity arise, it will be implemented anyway

Reply:
Doesn't it mean that the camera won't show objects further than 100.0? It would be pretty sad if someone's use case is a large scale Star-Wars like robot fleet.